src/realm/array_flex.cpp

```diff
@@ -159,29 +159,207 @@ bool ArrayFlex::find_all(const Array& arr, int64_t value, size_t start, size_t e
+    for (; start < end; ++start, ++it) {
+        if (*it == v_start)
+            continue;
+        if (!state->match(start + baseindex))
+            return false;
+    }
+
+    //    bf_iterator it{data, static_cast<size_t>(offset), ndx_width, ndx_width, start};
+    //    for (; start < end; ++start, ++it) {
+    //        const auto v = sign_extend_field_by_mask(mask, read_bitfield(data, (*it * v_width), v_width));
+    //        if (v != value)
+    //            if (!state->match(start + baseindex))
+    //                return false;
+    //    }
+    return true;
+}
+
+bool ArrayFlex::find_lt(const Array& arr, int64_t value, size_t start, size_t end, size_t baseindex,
+                        QueryStateBase* state) const
+{
+    const auto& encoder = arr.m_encoder;
     const auto v_width = encoder.m_v_width;
     const auto v_size = encoder.m_v_size;
     const auto ndx_width = encoder.m_ndx_width;
     const auto mask = encoder.width_mask();
+    const auto data = (uint64_t*)arr.m_data;
+    const auto offset = v_size * v_width;
+    // this can be made better... find GTE index and find all the matching values Less than that index..
+    //    auto v_start = parallel_subword_find<Less>(arr, value, encoder.m_v_mask, 0, v_width, 0, v_size);
+    //    start = parallel_subword_find<Equal>(arr, v_start, encoder.m_ndx_mask, offset, ndx_width, start, end);
+    //    bf_iterator it(data, offset, ndx_width, ndx_width, start);
+    //    for(; start<end; ++start,++it) {
+    //        if(*it > v_start) continue;
+    //        if(!state->match(start + baseindex))
+    //            return false;
+    //    }
 
-    auto cmp = [](int64_t v, int64_t value) {
-        if constexpr (std::is_same_v<Cond, Equal>)
-            return v == value;
-        if constexpr (std::is_same_v<Cond, NotEqual>)
-            return v != value;
-        if constexpr (std::is_same_v<Cond, Greater>)
-            return v > value;
-        if constexpr (std::is_same_v<Cond, Less>)
-            return v < value;
-    };
+    bf_iterator it_index{data, static_cast<size_t>(offset), ndx_width, ndx_width, start};
+    for (; start < end; ++start, ++it_index) {
+        const auto v = sign_extend_field_by_mask(mask, read_bitfield(data, *it_index * v_width, v_width));
+        if (v < value)
+            if (!state->match(start + baseindex))
+                return false;
+    }
+    return true;
+}
 
+bool ArrayFlex::find_gt(const Array& arr, int64_t value, size_t start, size_t end, size_t baseindex,
+                        QueryStateBase* state) const
+{
+    const auto& encoder = arr.m_encoder;
+    const auto v_width = encoder.m_v_width;
+    const auto v_size = encoder.m_v_size;
+    const auto ndx_width = encoder.m_ndx_width;
+    const auto mask = encoder.width_mask();
+    const auto data = (uint64_t*)arr.m_data;
     const auto offset = v_size * v_width;
+
+    // inverse logic of Less should be possible here.
```

Comment:
not quite because the order of scanning the values matter. For GT you'll need the index of the *last* value which is less or equal, then you can search for the index with GT.

This is different than LT, where you need the *first* index which is greater or equal, and then search with LT.